### PR TITLE
Used Bean Validations to validate workflows

### PR DESCRIPTION
**Pull request by hbelmiro**
By using Jakarta Bean Validations we automatically gain several validations and we can remove some of those manually implemented.

This PR: 

- Adds an invocation of Bean Validations to `WorkflowValidatorImpl`
- Removes existing manually implemented validations that are automatically done by Bean Validations
- Updates Java Bean Validations dependency to the re-branded Jakarta Bean Validations
- Includes Bean Validations implementation as dependency
  - Hibernate Validator
  - Glassfish Jakarta Expression Language (EL) Implementation
- Updates jsonschema2pojo Maven plugin to support Jakarta Bean Validations
- Fixes schemas to match https://serverlessworkflow.io/schemas/0.8/workflow.json for tests that started to fail after using Bean Validations

### CODE REVIEW

**Comment by fjtirado** [comment]
@tsurdilo Can you please take a look at this one?

File: pom.xml [hunk 1]
```diff
@@ -48,14 +48,14 @@
...
-        <version.javax.validation>2.0.1.Final</version.javax.validation>
+        <version.jakarta.validation>3.0.2</version.jakarta.validation>
         <version.org.junit.minor>6.0</version.org.junit.minor>
         <version.org.junit>5.${version.org.junit.minor}</version.org.junit>
         <version.org.junit.jupiter>${version.org.junit}</version.org.junit.jupiter>
         <version.org.mockito>3.0.0</version.org.mockito>
         <version.ch.qos.logback>1.1.3</version.ch.qos.logback>
         <version.org.assertj>3.13.2</version.org.assertj>
-        <version.jsonschema2pojo-maven-plugin>1.0.1</version.jsonschema2pojo-maven-plugin>
+        <version.jsonschema2pojo-maven-plugin>1.1.2 </version.jsonschema2pojo-maven-plugin>
```
Comment (tsurdilo) [inline pom.xml:58]: can this please be done as a separate pr? dont think its tied to the jakarta stuff? thanks

File: validation/src/test/java/io/serverlessworkflow/validation/test/WorkflowValidationTest.java [hunk 2]
```diff
@@ -90,7 +94,8 @@ public void testWorkflowMissingStates() {
     Assertions.assertNotNull(validationErrors);
     Assertions.assertEquals(1, validationErrors.size());
 
-    Assertions.assertEquals("No states found", validationErrors.get(0).getMessage());
+    Assertions.assertEquals(
+        "#/states: expected minimum item count: 1, found: 0", validationErrors.get(0).getMessage());
```
Comment (tsurdilo) [inline validation/src/test/java/io/serverlessworkflow/validation/test/WorkflowValidationTest.java:98]: This seems like json validation thats already available or am i missing something?

**Comment by tsurdilo** [comment]
Trying to understand the difference between this and what we already have available through json validation. Please let me know thanks.

**Comment by hbelmiro** [comment]
> Trying to understand the difference between this and what we already have available through json validation. Please let me know thanks.

This doesn't replace json validation, although we can if we want.
The reason to use Jakarta is to run validations that today are done manually. If you take a look at https://github.com/serverlessworkflow/sdk-java/pull/208/files#diff-93b6cea53affb42009f67eeb53b2b671c93b52717519932e2d26519b1c0c5f73 you'll see that several manually implemented validations were removed.

**Comment by hbelmiro** [comment]
@ricardozanini closing as stale.
